discovery/manager.go

```diff
@@ -0,0 +1,290 @@
+// Copyright 2016 The Prometheus Authors
```

Comment:
could you do a git mv here, otherwise it's very hard to follow the history of this file

Reply:
good point. I rearranged the commits and used git mv ,
Now when you click on the actual commits it shows it properly , 
https://github.com/prometheus/prometheus/pull/3362/commits/8b656714002bd01ae544cdd23453656f85d64c99 , but the final PR still shows as if the file is deleted and a new one is created.


the PR is just  to get some opinions and than I will arrange it properly and add some tests as well.